### ISSUE TITLE
Opening up the ServerDefinition to all languages

### ISSUE THREAD

**Issue by payne911** [ballerina-platform/lsp4intellij#339]
It would be great if we wouldn't have to exhaustively declare all supported file types in the ServerDefinition.

**Comment by nixel2007** [comment]
You can add a property in your plugin and make it configurable by end user

**Comment by payne911** [comment]
But then the end user would themselves have to list every single individual extension they want to support ?

**Comment by nixel2007** [comment]
Yes. Also there are file watches created for files with this extensions so plugin can handle file modifications and send events to the language server.
Why do you need to register all extensions? Is your language server a generic one, which eally can handle any file, and not tied to any programming language?

**Comment by payne911** [comment]
> Is your language server a generic one, which eally can handle any file, and not tied to any programming language?

No, but it's annoying to have to maintain a list of the 50+ extensions we support. The extensions we don't support are somewhat rare and usually wouldn't exist within our company's repository so it's safe to assume that it would indeed handle pretty much any file opened.

... but yes, obviously the better thing to do is indeed to list each extension.
